src/transforms/filter.js

```diff
@@ -8,211 +8,264 @@
 
 'use strict';
 
-var isNumeric = require('fast-isnumeric');
-
-// var Lib = require('@src/lib');
 var Lib = require('../lib');
+var axisIds = require('../plots/cartesian/axis_ids');
 
-/* eslint no-unused-vars: 0*/
+var INEQUALITY_OPS = ['=', '<', '>=', '>', '<='];
+var INTERVAL_OPS = ['[]', '()', '[)', '(]', '][', ')(', '](', ')['];
+var SET_OPS = ['{}', '}{'];
 
-// so that Plotly.register knows what to do with it
 exports.moduleType = 'transform';
 
-// determines to link between transform type and transform module
 exports.name = 'filter';
 
-// ... as trace attributes
 exports.attributes = {
+    enabled: {
+        valType: 'boolean',
+        dflt: true,
+        description: [
+            'Determines whether this filter transform is enabled or disabled.'
+        ].join(' ')
+    },
+    filtersrc: {
+        valType: 'enumerated',
+        values: ['x', 'y', 'z', 'ids'],
```

Comment:
Based on request and code sample from @etpinard, the `groupby` transform already uses the attribute crawler which had been extracted out of `plot_schema.js` and now lives in `Lib/coerce.js`. I was asked during the `groupby` PR review to elevate the crawler call to `transform` so I think there's plan to eventually use it in `filter`, maybe as a subsequent PR.